src/ext/serial.cpp

```diff
@@ -65,24 +118,41 @@ template<typename T> class ThingSetSerial: public ThingSetStream
  */
 
 #ifdef UART_SERIAL_ENABLED
-    extern const int pub_channel_serial;
-    extern Serial serial;
 
-    ThingSetSerial ts_uart(serial, pub_channel_serial);
+    #ifdef __MBED__
+        extern const int pub_channel_serial;
```

Comment:
Tiny remark: This one could be outside of #ifdef, as it is used in mbed and Zephyr.